### PR TITLE
[ParseableInterfaces] Add -prebuilt-module-cache-path to the frontend

### PR DESCRIPTION
**Pull request by jrose-apple**
When trying to load a swiftinterface, search this directory before doing all the work of building a swiftmodule.

### CODE REVIEW

File: lib/Frontend/ParseableInterfaceSupport.cpp [hunk 1]
```diff
@@ -472,31 +475,67 @@ std::error_code ParseableInterfaceModuleLoader::openModuleFiles(
     return std::make_error_code(std::errc::not_supported);
   }
 
-  // At this point we're either in PreferParseable mode or there's no credible
-  // adjacent .swiftmodule so we'll go ahead and start trying to convert the
-  // .swiftinterface.
+  // If we have a prebuilt cache path, check that too if the interface comes
+  // from the SDK.
+  if (!PrebuiltCacheDir.empty()) {
```
[the remark this answers is not in the document]
Comment (graydon) [reply lib/Frontend/ParseableInterfaceSupport.cpp]: I don't think the extra stats here are worth worrying about. Entire modules are comparatively large units of granularity; the system is doing a ton more stat-work examining individual file dependencies.

**Review by graydon — APPROVED** [review]
LGTM!

File: lib/Frontend/ParseableInterfaceSupport.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jrose-apple) [reply lib/Frontend/ParseableInterfaceSupport.cpp]: It's less the stats I'm worried about than the dependency checking, i.e. "the prebuilt version is present but I've modified my SDK for whatever reason". Or even, to a lesser extent, "the prebuilt version is present but I'm using a custom-built toolchain".

**Comment by jrose-apple** [comment]
@swift-ci Please test Linux

**Review by harlanhaskins — APPROVED** [review]

**Review by nathawes — APPROVED** [review]
👍

**Comment by jrose-apple** [comment]
I discovered a silly bug: this works at the top level, but I forgot to pass it down to the sub-invocation to handle indirect imports. Fix coming soon.

**Comment by jrose-apple** [comment]
Merged as part of #21513.

**Pull request merged**